Foundation/NSString.swift

```diff
@@ -1234,8 +1231,8 @@ extension NSString {
     }
     
     public convenience init?(bytes: UnsafeRawPointer, length len: Int, encoding: UInt) {
-        let bytePtr = bytes.bindMemory(to: UInt8.self, capacity: len)
-        guard let cf = CFStringCreateWithBytes(kCFAllocatorDefault, bytePtr, len, CFStringConvertNSStringEncodingToEncoding(encoding), true) else {
+        let extRep = NSData(bytesNoCopy: UnsafeMutableRawPointer(mutating: bytes), length: len, freeWhenDone: false)._cfObject
```

Comment:
I'm not quite sure I understand what you mean by placeholder string. However, it's clear that Apple Foundation reads past the internal null character with `NSString.init(bytes:length:encoding:)`, as demonstrated below:

```bash
echo -e "This is a string.\0This is another string." > /Users/xwu/Desktop/test.txt
```

```swift
let url = URL(fileURLWithPath: "/Users/xwu/Desktop/test.txt")
let data = try Data(contentsOf: url)
data.withUnsafeBytes { (bytes: UnsafePointer<UInt8>) in
  let string = NSString(bytes: bytes, length: data.count, encoding: String.Encoding.ascii.rawValue)
  // "This is a string.\0This is another string."
}
```

Not sure what it would take to achieve this result if we are to use `CFStringCreateWithBytes` in corelibs-foundation. Likewise with the original bug regarding `init(contentsOf: URL, encoding: UInt)`: I'm not sure how to interpret the discrepancy if Apple Foundation is doing nothing more than calling through to `CFStringCreateWithBytes`.